src/btree/bt_split.c

```diff
@@ -405,8 +383,7 @@ __split_deepen(WT_SESSION_IMPL *session, WT_PAGE *parent)
 	 * Create N children, unless we are dealing with a large page without
 	 * many entries, in which case split into the minimum number of pages.
 	 */
-	children = WT_MAX(pindex->entries / __split_deepen_per_child,
-	    __split_deepen_min_child);
+	children = pindex->entries / __split_deepen_per_child;
```

Comment:
Why this change: if we had `memory_page_max` set to 1MB and 100KB keys, we could still end up with `children == 0` here?